r2r/pipelines/basic/scraping.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from r2r.core import ScraperPipeline
+from r2r.core import BasicDocument
+
+from typing import Iterator, Optional
+from bs4 import BeautifulSoup
+import requests
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_raw_html_from_url(url: str) -> str:
+    try:
+        response = requests.get(url)
+
+        if response.status_code == 200:
+            return response.text
+        else:
+            return (
+                f"Failed to retrieve HTML. Status code: {response.status_code}"
+            )
+    except requests.exceptions.RequestException as e:
+        return f"An error occurred: {e}"
+
+
+class BasicScraperPipeline(ScraperPipeline):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def scrape_url(self, url: str) -> Iterator[BasicDocument]:
```

Comment:
Did we decide to not use scrapy? This is fine for a very basic implementation, but in my experience this approach will fail with a reasonably high rate, whereas scrapy can help mitigate that.